lib/chef/rest.rb

```diff
@@ -188,9 +370,25 @@ def follow_redirect
 
     public :create_url
 
+    def create_url(path)
+      return path if path.is_a?(URI)
+      if path =~ /^(chefzero):\/\//i
+        URI.parse(path)
+      else
+        super
+      end
+    end
+
     def http_client(base_url=nil)
       base_url ||= url
-      BasicClient.new(base_url, :ssl_policy => Chef::HTTP::APISSLPolicy)
+      pp url_class: base_url.class, value: base_url
```

Comment:
Yeah, this extremely rough, I only put it up here to let people vet the design. I'm certain there are a bunch of failing integration tests (which travis might not run so I'm not sure if it'll pass travis, but it's broken anyway).